services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/CacheService.java

```diff
@@ -60,19 +64,19 @@
      */
     private FlowCache flowCache;
 
-    public CacheService(NetworkCache networkCache, FlowCache flowCache, PathComputer pathComputer) {
+    public CacheService(NetworkCache networkCache, FlowCache flowCache, RepositoryFactory repositoryFactory) {
         this.networkCache = networkCache;
         this.flowCache = flowCache;
 
         logger.info("Request initial network state");
 
-        initFlowCache(pathComputer);
-        initNetwork(pathComputer);
+        initFlowCache(repositoryFactory.createFlowRepository());
+        initNetwork(repositoryFactory.createSwitchRepository(), repositoryFactory.createIslRepository());
     }
 
-    private void initFlowCache(PathComputer pathComputer) {
+    private void initFlowCache(FlowRepository flowRepository) {
         logger.info("Flow Cache: Initializing");
-        PathComputerFlowFetcher flowFetcher = new PathComputerFlowFetcher(pathComputer);
+        BidirectionalFlowFetcher flowFetcher = new BidirectionalFlowFetcher(flowRepository);
```

Comment:
It raise `InvalidArgumentException` in case of corrupted data in DB... as result of exception here - storm will not start whole topology.

Reply:
No, it doesn't. As you mentioned above we catch and log IllegalArgumentException.